src/connector/src/source/cdc/external/mod.rs

```diff
@@ -208,12 +208,14 @@ impl MySqlOffset {
     }
 }
 
+pub type CdcOffsetParseFunc = Box<dyn Fn(&str) -> ConnectorResult<CdcOffset> + Send>;
+
 pub trait ExternalTableReader {
     fn get_normalized_table_name(&self, table_name: &SchemaTableName) -> String;
 
     async fn current_cdc_offset(&self) -> ConnectorResult<CdcOffset>;
 
-    fn parse_cdc_offset(&self, dbz_offset: &str) -> ConnectorResult<CdcOffset>;
+    fn get_cdc_offset_parser(&self) -> CdcOffsetParseFunc;
```

Comment:
None of the 3 implementation uses `self` actually. Please remove `self` to make it a tpye-associated function.

Reply:
Without the `self`, the `enum ExternalTableReaderImpl` cannot match `self` to know the specific variant. I remove this function from the trait.